internal/services/groups/groups.go

```diff
@@ -42,6 +45,45 @@ func groupFindByName(ctx context.Context, client *msgraph.GroupsClient, displayN
 	return &result, nil
 }
 
+func isGroupInAdministrativeUnit(ctx context.Context, client *msgraph.GroupsClient, auId string, id string) (bool, int, error) {
+	resp, status, _, err := client.BaseClient.Get(ctx, msgraph.GetHttpRequestInput{
+		ConsistencyFailureFunc: msgraph.RetryOn404ConsistencyFailureFunc,
+		OData: odata.Query{
+			Select: []string{"id"},
+		},
+		ValidStatusCodes: []int{http.StatusOK},
+		Uri: msgraph.Uri{
+			Entity:      fmt.Sprintf("/groups/%s/memberOf/microsoft.graph.administrativeUnit", id),
+			HasTenantId: true,
+		},
+	})
+	if err != nil {
+		return false, status, fmt.Errorf("GroupsClient.BaseClient.Get(): %v", err)
+	}
+	defer resp.Body.Close()
+	respBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return false, status, fmt.Errorf("io.ReadAll(): %v", err)
+	}
+
+	var data struct {
+		Members []struct {
+			Id string `json:"id"`
+		} `json:"value"`
+	}
+	if err := json.Unmarshal(respBody, &data); err != nil {
+		return false, status, fmt.Errorf("json.Unmarshal(): %v", err)
+	}
+
+	for _, v := range data.Members {
+		if v.Id == auId {
+			return true, status, nil
+		}
+	}
+
+	return false, status, nil
+}
+
```

Comment:
As per above, I don't _think_ this function will be needed if we look up by group membership rather than per AU?

Reply:
I think I don't understand what you mean here.
I can see two ways to get the information, whether a group is member of an AU:
1. List the "member of" of a group: https://learn.microsoft.com/en-us/graph/api/group-list-memberof?view=graph-rest-1.0&tabs=http
2. List the mebers of an AU: https://learn.microsoft.com/en-us/graph/api/administrativeunit-list-members?view=graph-rest-1.0&tabs=http

At the moment we're using option 1, are you suggesting to use option 2 instead?
I think from a permissions point of view, option 2 requires less privileges than option 1 (GroupMember.Read.All for option 1,  AdministrativeUnit.Read.All for option 2).

Reply:
Sorry I seem to have misread this. Looking up by group membership feels like the right way to accomplish this, however I would amend this function to return a []string containing AU IDs (then iterating over that slice with a local function). And by doing that, there's no reason why this shouldn't make it upstream into the SDK as part of the GroupsClient.

e.g.

```go

// PROVIDER

isGroupInAdministrativeUnit := func(aus *[]msgraph.AdministrativeUnit, id string) bool {
	for _, au := range *aus {
		if au.ID == id {
			return true
		}
	}
	return false
}


// SDK

func (*GroupsClient) ListAdministrativeUnitMemberships(ctx context.Context, id string) (*[]AdministrativeUnit, int, error) {
	// ...
	return &[]AdministrativeUnit{}
}
```

Hope that makes sense?

Reply:
I have created this PR in hamilton:
https://github.com/manicminer/hamilton/pull/220
We don't need to check if the group is part of the au IDs that are part of the terraform configuration anymore, as now the `administrative_unit_ids` is declarative.